### ISSUE TITLE
compilation issue in calculations.cpp

### ISSUE THREAD

**Issue by ryzhyk** [p4lang/behavioral-model#65]
I am getting the following compilation error with gcc 5.2.1, which can be suppressed by `./configure CXXFLAGS="-Wno-error"`

```
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I../.. -DBMELOG_ON -I../../modules/bf_lpm_trie/include -isystem../../third_party/jsoncpp/include -isystem../../third_party/spdlog/include -I../../modules/BMI/include -I./include/ -DBMDEBUG_ON -Wall -Werror -Wextra -g -O2 -std=c++11 -MT src/calculations.lo -MD -MP -MF src/.deps/calculations.Tpo -c src/calculations.cpp  -fPIC -DPIC -o src/.libs/calculations.o
In file included from src/calculations.cpp:26:0:
src/calculations.cpp:68:15: error: 'bm::{anonymous}::xxh64_create_' defined but not used [-Werror=unused-variable]
 REGISTER_HASH(xxh64);
               ^
./include/bm_sim/calculations.h:400:8: note: in definition of macro 'REGISTER_HASH'
   bool hash_name##_create_ =                                            \
        ^
src/calculations.cpp:88:15: error: 'bm::{anonymous}::crc16_create_' defined but not used [-Werror=unused-variable]
 REGISTER_HASH(crc16);
               ^
./include/bm_sim/calculations.h:400:8: note: in definition of macro 'REGISTER_HASH'
   bool hash_name##_create_ =                                            \
        ^
src/calculations.cpp:102:15: error: 'bm::{anonymous}::crc32_create_' defined but not used [-Werror=unused-variable]
 REGISTER_HASH(crc32);
               ^
./include/bm_sim/calculations.h:400:8: note: in definition of macro 'REGISTER_HASH'
   bool hash_name##_create_ =                                            \
        ^
src/calculations.cpp:116:15: error: 'bm::{anonymous}::crcCCITT_create_' defined but not used [-Werror=unused-variable]
 REGISTER_HASH(crcCCITT);
               ^
./include/bm_sim/calculations.h:400:8: note: in definition of macro 'REGISTER_HASH'
   bool hash_name##_create_ =                                            \
        ^
src/calculations.cpp:164:15: error: 'bm::{anonymous}::cksum16_create_' defined but not used [-Werror=unused-variable]
 REGISTER_HASH(cksum16);
               ^
./include/bm_sim/calculations.h:400:8: note: in definition of macro 'REGISTER_HASH'
   bool hash_name##_create_ =                                            \
        ^
src/calculations.cpp:172:15: error: 'bm::{anonymous}::csum16_create_' defined but not used [-Werror=unused-variable]
 REGISTER_HASH(csum16);
               ^
./include/bm_sim/calculations.h:400:8: note: in definition of macro 'REGISTER_HASH'
   bool hash_name##_create_ =                                            \
        ^
src/calculations.cpp:185:15: error: 'bm::{anonymous}::identity_create_' defined but not used [-Werror=unused-variable]
 REGISTER_HASH(identity);
               ^
./include/bm_sim/calculations.h:400:8: note: in definition of macro 'REGISTER_HASH'
   bool hash_name##_create_ =                                            \
        ^

```

**Comment by antoninbas** [comment]
I will try to see if I can fix this. In the meantime, you can compile it with one of the g++ version we support: 4.8, 4.9 or 5.0

**Comment by antoninbas** [comment]
I isolated the issue and fixed it in #67 
Note that the issue was only present with g++5.2, there was no issue with g++5.3. I added a g++5.2 regression to the Travis build.

**Comment by ryzhyk** [comment]
Just confirming: it works for me too. Many thanks for the fix!
